### DISCUSSION TITLE
ServerJobFilter related questions

### DISCUSSION THREAD

**Discussion by JoonasC** [jobrunr/jobrunr#994] [Q&A] [answered]
The JavaDoc for `ServerJobFilter` states that all classes that implement it should run really fast, or they will be removed in order to preserve performance. What is the algorithm used to determine when a `ServerJobFilter` is slow? I'm considering putting notification functionality into a `ServerJobFilter`, and it would be really bad if a notification for a completed job does not get sent. Is `ServerJobFilter` the best tool for this use-case, or are there other ways to implement it? Also, do all servers participating in the JobRunr cluster get notified via `ServerJobFilter` when a job succeeds/fails, or does this happen only for the server that submitted the job?

**Comment by auloin** [comment, marked as answer]
Hi @JoonasC, the `JobFilter` will not be removed, we only forgot to update the JavaDoc, but thanks to you that'll be fixed soon.

> What is the algorithm used to determine when a ServerJobFilter is slow?

Currently we consider the `JobFilter` to be slow if it takes more than 10ms to run, the message is logged if that's the case.

> Is ServerJobFilter the best tool for this use-case

Yes that's the best option in OSS.

> does this happen only for the server that submitted the job?

Only the server on which the job is executed sends such an event others are not notified.

**Reply by JoonasC** [reply]
> Only the server on which the job is executed sends such an event others are not notified

This is a bit problematic for me, as I would need to have all the servers in my cluster get notified. I'm guessing this is not supported in OSS?

**Reply by rdehuyss** [reply]
I'm afraid not, as we've heard many times that JobRunr OSS is feature complete, we still try to have some features that help companies to get a Pro license...

We still have kids to feed 🍼 😊.
